consensus/src/detail/mod.rs

```diff
@@ -202,6 +203,34 @@ impl Consensus {
         let block = consensus_ref.db_tx.get_block(id).map_err(BlockError::from)?;
         Ok(block)
     }
+
+    pub fn get_header_from_height(
+        &self,
+        height: &BlockHeight,
+    ) -> Result<Option<BlockHeader>, BlockError> {
+        let id = self.get_block_id_from_height(height)?.ok_or(BlockError::NotFound)?;
+        Ok(Some(
+            self.get_block(id)?.ok_or(BlockError::NotFound)?.header().clone(),
```

Comment:
I think we can come up with a way of only returning headers directly from the db. Similarly as we do for transactions, just the offset is always 1 and decode the appropriate bit.

Reply:
I'll try that, thanks

Reply:
Use the block index

Reply:
@TheQuantumPhysicist could you elaborate?

I think the blockheaders are variable-sized because they contain `Option<T>` fields so isn't it true we'd have to store the serialized size of them somewhere in order to query them from the storage? For now I just read the entire block and only return header which is a small API improvement.

Reply:
The block index should either contain the header size or the whole header. In bitcoin it contains the whole header. Reading the entire block just for the header is crazy.

Reply:
Yeah it's idiotic, I'll try storing the header to block index and see how well that works. Thanks

Reply:
Btw, I think Roy is doing the same as well. I thought it's in master already, but it's not.

Reply:
Sweet, I'll rebase on top of his branch then. Thanks again

Reply:
I'm not sure when he's gonna merge. If you're in hurry, it's OK to beat him up to it to master.

Reply:
Nope, not in a rush